### ISSUE TITLE
Link to user's own profile on remote pumps

### ISSUE THREAD

**Issue by shtrom** [e14n/pump.io#1102]
When clicking on the user's name on the top-right corner, we get two different behaviours depending on whether this is the user's own pump, or a remote pump:
* locale pump: links to activity, favourites, ...; settings, account; logout
* remote pump: logout (only)

It would be good to unify those a bit, adding the links to activity, favourites, ... on the remote pump interface too (I'm assuming they can be derived from the user's information, correct me if I'm wrong). Settings and account might however not make sense on a remote pump, though.

Also, it would be good to add a link to get back to the user's home stream. On the local pump, they just have to click the network's name on the top-left, but this leads to the local server stream, rather than home stream, on remote servers.

A link reliably taking the user back to their home view, say, from the top-right menu, would probably ease navigation.

**Comment by strugee** [comment]
I think we have to ask ourselves how we want to present the web app. Do we want each site to present its web interface as a separate app? Do we want all web interfaces to look the same, so they feel like one web app? Seems to me this will be _extremely_ hard to accomplish technically.

**Comment by shtrom** [comment]
On Sun, Oct 04, 2015 at 06:52:24PM -0700, Alex Jordan wrote:
> I think we have to ask ourselves how we want to present the web app.
> Do we want each site to present its web interface as a separate app?
> Do we want all web interfaces to look the same, so they feel like one
> web app? Seems to me this will be _extremely_ hard to accomplish
> technically.

Hum, that's a fair point (thinking of, e.g., @jpope777's pump). I think Evan
has repeatedly said that the Webapp was just a bootstrap mechanisms, and
that he was expecting people to later use other applications.

Perhaps the webapp should consume streams from remote site, without ever
accessing the HTTP URLs themselves, so that everything would be
displayed within the home webapp.

That said, the current web frontend is still a default upon which other
web apps are tweaked (see @jpope777's again, or steve@hub.polari.us's),
so fixing this there is likely to trickle down to other forks.

-- 
Olivier Mehani <shtrom@ssji.net>
PGP fingerprint: 4435 CF6A 7C8D DD9B E2DE  F5F9 F012 A6E2 98C6 6655
Confidentiality cannot be guaranteed on emails sent or received unencrypted.

**Comment by strugee** [comment]
Yeah, I was thinking of jpope's pump, too. IIRC that felt like a separate application, just because it looked so different.

By this:

> Perhaps the webapp should consume streams from remote sites, without ever accessing the HTTP URLs themselves, so that everything would be displayed within the home webapp.

I assume you mean doing something similar to what diaspora* does? It could be done, but it seems like a lot of work and also, doesn't feel very "webby" to me.

**Comment by cwebber** [comment]
I think there's two things here:

1) How much we should "help" users on a remote pump using the web client.  I think if we can help users navigate back and around, that's good.  We don't want users getting lost.  On the other hand, in a sense this is slightly dangerous: a remote server can "lie" about what those buttons are and if users get used to this, this could be a great way to phish users into doing activities they didn't intend.  I worry about this somewhat.  If we start doing this, we need a very clear visual indicator that "you are in remote mode".  But helping users get back can help them from getting lost.  So maybe what we *really* want is a "you are in remote mode", and that indicator of remote mode is something clickable that takes you back to your "home" server.  That's simple enough... users can navigate to their other tabs and etc then.
2) The amount we should spend investing in the web client.  Yes, the web client is "just another client", but the value here is high; the web interface is the main way that users interact with the pumpiverse, and the low polish on the web application is something that pushes many users away.

**Comment by strugee** [comment]
@cwebber AFAIK, remote pumps wouldn't even have to lie about buttons, right? If the remote pump is malicious, it's game over, no matter what we do. Any indicator about "remoteness" will be patched out by the malicious actor.

Don't get me wrong, I'm worried about this too. I'd like to find a solution that was secure but also had the "webbiness" of the currrent scheme, but right now, I don't see one.

**Comment by strugee** [comment]
That being said, yeah, we should fix the web client. Even just fixing some of the obvious bugs would go a long way.
